gneiss/regression/_mixedlm.py

```diff
@@ -0,0 +1,270 @@
+
+    Here, we will define a table of proportions with 3 features
+    `a`, `b`, and `c` across 12 samples.
+
+    >>> table = pd.DataFrame({
+    ...         'u1':  [0.804248, 0.195526, 0.000226],
+    ...         'u2':  [0.804369, 0.195556, 0.000075],
+    ...         'u3':  [0.825711, 0.174271, 0.000019],
+    ...         'x1':  [0.751606, 0.158631, 0.089763],
+    ...         'x2':  [0.777794, 0.189095, 0.033111],
+    ...         'x3':  [0.817855, 0.172613, 0.009532],
+    ...         'y1':  [0.780774, 0.189819, 0.029406],
+    ...         'y2':  [0.797332, 0.193845, 0.008824],
+    ...         'y3':  [0.802058, 0.194994, 0.002948],
+    ...         'z1':  [0.825041, 0.174129, 0.000830],
+    ...         'z2':  [0.804248, 0.195526, 0.000226],
+    ...         'z3':  [0.825667, 0.174261, 0.000072]}
+    ...         index=['a', 'b', 'c']).T
+
+    Now we are going to define some of the external variables to
+    test for in the model.  Here we will be testing a hypothetical
+    longitudinal study across 3 time points, with 4 patients
+    `x`, `y`, `z` and `u`, where `x` and `y` were given treatment `1`
+    and `z` and `u` were given treatment `2`.
+
+    >>> metadata = pd.DataFrame({
+    ...         'patient': [1, 1, 1, 2, 2, 2, 3, 3, 3, 4, 4, 4],
+    ...         'treatment': [1, 1, 1, 1, 1, 1, 2, 2, 2, 2, 2, 2],
+    ...         'time': [1, 2, 3, 1, 2, 3, 1, 2, 3, 1, 2, 3]
+    ...     }, index=['x1', 'x2', 'x3', 'y1', 'y2', 'y3',
+    ...               'z1', 'z2', 'z3', 'u1', 'u2', 'u3'])
+
+    Finally, we need to define a bifurcating tree used to convert the
+    proportions to balances.  If the internal nodes aren't labels,
+    a default labeling will be applied (i.e. `y1`, `y2`, ...)
+
+    >>> tree = TreeNode.read(['(c, (b,a)Y2)Y1;'])
+    >>> print(tree.ascii_art())
+              /-c
+    -Y1------|
+             |          /-b
+              \Y2------|
+                        \-a
+
+    Now we can run the linear mixed effects model on the proportions.
+    Underneath the hood, the proportions will be transformed into balances,
+    so that the linear mixed effects models can be run directly on balances.
+    Since each patient was sampled repeatedly, we'll specify them separately
+    in the groups.  In the linear mixed effects model `time` and `treatment`
+    will be simultaneously tested for with respect to the balances.
+
+    >>> res = mixedlm('time + treatment', table, metadata, tree,
+    ...               groups='patient')
+
+    See Also
+    --------
+    statsmodels.regression.linear_model.MixedLM
+    skbio.stats.composition.multiplicative_replacement
+    ols
+
+    """
+    table, metadata, tree = _intersect_of_table_metadata_tree(table,
+                                                              metadata,
+                                                              tree)
+    ilr_table, basis = _to_balances(table, tree)
+    data = pd.merge(ilr_table, metadata, left_index=True, right_index=True)
```

Comment:
I often have the problem that indices for metadata and the counts have different data type, e.g. int and str. Could you raise an Error if there is no overlap when merging?